File: CommonFiles/Common_Files_VB_Test/TestHelpers.cs
using static MetalUp.Express.Helpers;

namespace CommonFiles_VB_Test
{
    [TestClass]
    public class TestHelpers
    {
        #region Display
        [TestMethod]
        public void TestDisplayString()
        {
            Assert.AreEqual("\"Foo\"", Display("Foo"));
        }
        [TestMethod]
        public void TestDisplayInt()
        {
            Assert.AreEqual("1", Display(1));
        }
        [TestMethod]
        public void TestDisplayDouble()
        {
            Assert.AreEqual("3.3", Display(3.3));
        }
        [TestMethod]
        public void TestDisplayBool()
        {
            Assert.AreEqual("true", Display(true));
        }
        [TestMethod]
        public void TestDisplayListOfInt()
        {
            var actual = Display(new List<int> { 1, 2, 3 });
            Assert.AreEqual("\n{1, 2, 3}", actual );
        }
        [TestMethod]
        public void TestDisplayListOfString()
        {
            Assert.AreEqual("\n{\"foo\", \"bar\"}", Display(new List<string> { "foo", "bar" }));
        }
        [TestMethod]
        public void TestDisplayListOfLists()
        {
            Assert.AreEqual("\n{\n{1, 2, 3}, \n{4, 5}}", Display(new List<List<int>> { new List<int> { 1, 2, 3 }, new List<int> { 4, 5 } }));
        }
        [TestMethod]
        public void TestDisplayTuplet()
        {
            Assert.AreEqual("(\"foo\", \"bar\")", Display(("foo", "bar")));
        }
#endregion

        #region ArgString
        [TestMethod]
        public void TestArgString()
        {
            Assert.AreEqual("3, \"foo\", (1, \"bar\"), \n{1, 2, 3}", ArgString(3, "foo", (1, "bar"), new List<int> { 1, 2, 3 }));
        }
        #endregion
    }
}

File: CommonFiles/Common_Files_PY/TestHelpers.py
from Wrapper import *

import unittest

class TestHelpers(unittest.TestCase):

        #region Display

        def test_DisplayString(self):        
            self.assertEqual("'Foo'", display("Foo"))
            
        def test_DisplayInt(self):      
            self.assertEqual("1", display(1))
          
        def test_DisplayDouble(self):       
            self.assertEqual("3.3", display(3.3))
             
        def test_DisplayBool(self):       
            self.assertEqual("True", display(True))
            
        def test_DisplayListOfInt(self):       
            self.assertEqual("\n[1, 2, 3]", display([1, 2, 3 ]))
             
        def test_DisplayListOfString(self):      
            self.assertEqual("\n['foo', 'bar']", display(['foo', 'bar']))
            
        def test_DisplayListOfLists(self):      
            self.assertEqual("\n[\n[1, 2, 3], \n[4, 5]]", display([[1, 2, 3 ],[4, 5]]))
           
        def test_DisplayTuplet(self):
            self.assertEqual("('foo', 'bar')", display(("foo", "bar")))        
        #endregion

        #def test_ArgString(self):       
        #    self.assertEqual("", arg_string([3, "foo", (1, "bar"), [1,2,3] ]))
        
        def test_FailMessage(self):       
            self.assertEqual("xxxTest failed calling Foo(3, 4) Expected: 1 Actual: 2xxx", fail_message("Foo", 1, 2, [3, 4]))
        
        def test_EqualIfRounded(self):
        
            self.assertTrue(EqualIfRounded(3.456, 3.4562789))
            self.assertFalse(EqualIfRounded(3.4562, 3.4562789))
        
        #def test_SetItem(self):      
        #    inp = [1, 2, 3, 4, 5]
        #    expected = [1, 2, 6, 4, 5]
        #    self.assertEqual(expected, inp.SetItem(2, 6))
        
        #def test_InsertItem(self):       
        #    inp =[1, 2, 3, 4, 5 ]
        #    expected = [ 1, 2, 6, 3, 4, 5 ]
        #    self.assertEqual(expected, inp.InsertItem(2, 6))      
      
        #def test_RemoveItem(self):       
        #    inp =[1, 2, 3, 4, 5 ]
        #    expected = [ 1, 2, 4, 5 ]
        #    self.assertEqual(expected, inp.RemoveItem(2))
  
if __name__ == '__main__':
    unittest.main()

